projects/rocprofiler-sdk/source/include/rocprofiler-sdk/experimental/thread-trace/core.h

```diff
@@ -36,6 +36,20 @@ ROCPROFILER_EXTERN_C_INIT
  * @{
  */
 
+/**
+ * @brief Types of Thread Trace buffering  modes
+ * ::ROCPROFILER_THREAD_TRACE_PARAMETER_BUFFERING_MODE_NONE defines no preference
+ * TRIPLE_BUFFER is currently only available for a single shader engine at a time.
+ *
+ */
+typedef enum rocprofiler_thread_trace_parameter_buffering_mode_t
+{
+    ROCPROFILER_THREAD_TRACE_PARAMETER_BUFFERING_MODE_NONE = 0,
+    ROCPROFILER_THREAD_TRACE_PARAMETER_BUFFERING_MODE_SINGLE_BUFFER,
+    ROCPROFILER_THREAD_TRACE_PARAMETER_BUFFERING_MODE_TRIPLE_BUFFER,
```

Comment:
I'm fine with this for now. I'm just trying to think whether we should be have some more generic approach with the `ROCPROFILER_THREAD_TRACE_PARAMETER_BUFFERING_MODE_MULTIPLE_BUFFERS`, and then having an argument passed to thread trace that would configure the number of CPU/GPU buffers?

Also, maybe a nit-picking, maybe a plural is better.
```suggestion
    ROCPROFILER_THREAD_TRACE_PARAMETER_BUFFERING_MODE_TRIPLE_BUFFERS,
```

Reply:
We can add more options here later. My concern with allowing generic N buffers from the API is that we need to test a lot of possibilities.

Reply:
Normally I see the technique called "double/triple buffer" or "double/triple buffering", but not plural.